# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1419924254 -3600
#      Tue Dec 30 08:24:14 2014 +0100
# Branch gui-release
# Node ID 9c719c504e8d03bf3b14185a4ffd8e7529addd7a
# Parent  d57edbd761aa28e43625ff57a90205bb63d1286e
command window buffer size configurable on windows (bug #41474)

* QWinTerminalImpl.cpp (QConsolePrivate::setScrollBufferSize):
  changes the buffer size;
  (QWinTerminalImpl::setScrollBufferSize): function called from the routine
  that is reading new user preferences, just calls function above

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -611,20 +611,22 @@ void QConsolePrivate::setSelectionColor 
 void QConsolePrivate::setCursorColor (bool useForegroundColor,
                                       const QColor& color)
 {
   m_cursorColor = useForegroundColor ? QColor () : color;
 }
 
 void QConsolePrivate::setScrollBufferSize (int value)
 {
-// FIXME: not tested code follows
-//  CONSOLE_SCREEN_BUFFER_INFO sbi;
-//  GetConsoleScreenBufferInfo (m_stdOut, &sbi);
-//  m_bufferSize = QSize (sbi.dwSize.X, qMax (sbi.dwSize.Y, (SHORT)value));
+  CONSOLE_SCREEN_BUFFER_INFO sbi;
+  GetConsoleScreenBufferInfo (m_stdOut, &sbi);
+
+  m_bufferSize = QSize (sbi.dwSize.X, (SHORT)value);
+
+  updateConsoleSize (true);
 }
 
 void QConsolePrivate::drawTextBackground (QPainter& p, int cx1, int cy1,
                                           int cx2, int cy2, int cw, int ch)
 {
   p.save ();
 
   int ascent = p.fontMetrics ().ascent ();
@@ -1728,18 +1730,17 @@ void QWinTerminalImpl::setSelectionColor
 void QWinTerminalImpl::setCursorColor (bool useForegroundColor,
                                        const QColor& color)
 {
   d->setCursorColor (useForegroundColor, color);
 }
 
 void QWinTerminalImpl::setScrollBufferSize(int value)
 {
-// FIXME: not tested code follows
-//  d->setScrollBufferSize (value);
+  d->setScrollBufferSize (value);
 }
 
 
 //////////////////////////////////////////////////////////////////////////////
 
 void QWinTerminalImpl::setTerminalFont (const QFont& f)
 {
   d->m_font = f;
